sklearn/gaussian_process/gpr.py

```diff
@@ -261,6 +261,10 @@ def predict(self, X, return_std=False, return_cov=False):
             Covariance of joint predictive distribution a query points.
             Only returned when return_cov is True.
         """
+        if self.kernel is None:
```

Comment:
You need to create a local variable `kernel` if `self.kernel` is None as mentioned by @amueller .

And it might make sense to move it under the `if not hasattr(self, "X_train_"):` block